### PR TITLE
Unicode 16 first cut security data

### PR DESCRIPTION
**Pull request by markusicu**
- generate files
- basic Identifier_Type invariant tests
- bug fix: ID_Type Limited_Use trumps Exclusion
- ID_Type(A9CF)=Limited_Use Uncommon_Use (UTC-179-C39)
- set new characters to Uncommon_Use if they were initially generated as Recommended
- add to removals.txt criteria for Recommended vs. Uncommon_Use

Best reviewed one commit at a time.

### CODE REVIEW

**Review by Ken-Whistler — COMMENTED** [review]
What I could understand looks fine. I'm not familiar with the invariants code or some of the other internal data files here.

**Comment by macchiati** [comment]
I assume this will be modified further as a result of the conversation about the default IDNA type/status

**Comment by markusicu** [comment]
@dwanders-A & SAH/SEW: should any of the following **not be** Uncommon_Use? I am sending more details via email.

```
1C89..1C8A    ; Uncommon_Use                   # 16.0   [2] CYRILLIC CAPITAL LETTER TJE..CYRILLIC SMALL LETTER TJE

A7CB..A7CD    ; Uncommon_Use                   # 16.0   [3] LATIN CAPITAL LETTER RAMS HORN..LATIN SMALL LETTER S WITH DIAGONAL STROKE
A7DA..A7DC    ; Uncommon_Use                   # 16.0   [3] LATIN CAPITAL LETTER LAMBDA..LATIN CAPITAL LETTER LAMBDA WITH STROKE

10EC2..10EC4  ; Uncommon_Use                   # 16.0   [3] ARABIC LETTER DAL WITH TWO DOTS VERTICALLY BELOW..ARABIC LETTER KAF WITH TWO DOTS VERTICALLY BELOW

116D0..116E3  ; Uncommon_Use                   # 16.0  [20] MYANMAR PAO DIGIT ZERO..MYANMAR EASTERN PWO KAREN DIGIT NINE
```

File: unicodetools/data/security/dev/data/source/removals.txt [hunk 1]
```diff
@@ -794,3 +813,7 @@ A7F5; Obsolete
 A7F6; Obsolete
 AB68; Technical
 
+# Unicode 16
+# UTC-179-Cxx: Change the Identifier_Type of A9CF to Limited_Use Uncommon_Use, removing Exclusion.
```
[the remark this answers is not in the document]
Comment (markusicu) [reply unicodetools/data/security/dev/data/source/removals.txt]: done tnx

**Comment by macchiati** [comment]
@dwanders-A

The question should have been: Are they in **common/customary** modern use? And whenever we are in doubt, the default should be Uncommon_Use.

If it is in modern use, but very infrequently, such as in technical documents, then we need to know that too. For example, just from the https://en.wikipedia.org/wiki/Latin_gamma we have that 

`A lowercase Latin gamma that lies above the [baseline](https://en.wikipedia.org/wiki/Baseline_(typography)) rather than crossing it (ɤ, called "ram's horns"), represents the [close-mid back unrounded vowel](https://en.wikipedia.org/wiki/Close-mid_back_unrounded_vowel). In certain [nonstandard variations of the IPA](https://en.wikipedia.org/wiki/Obsolete_and_nonstandard_symbols_in_the_International_Phonetic_Alphabet) the uppercase form is used.`

What that indicate​s is that it should be marked as either **Uncommon_Use** or **Technical**. 

**It is important to note** whenever there is some doubt, we should default it to **Uncommon_Use** unless we have reasonable evidence that it is in common use. This data does not affect the use of the character for normal purposes — writing books, articles, text messages, and so on; it is specially designed for identifiers and similar constructs.

We can **always** set "upgrade" it later on ​w​henever someone ​p​resents a reasonable case for it being in common/customary use. For example,​ if ​s​omeone finds that the uppercase form is used in the normal orthography for a modern language X, which has significant, active population using it​, and makes a proposal to that effect, that would be justification for dropping the Identifier_Type of  Uncommon_Use or Technical.

**Comment by markusicu** [comment]
@macchiati @asmusf et al. -- no new Recommended characters; could you (or someone) please approve this PR?

**Review by macchiati — APPROVED** [review]
Looks good as a first cut

**Review by asmusf — COMMENTED** [review]
Seems fine

**Pull request merged**